rcl_action/CMakeLists.txt

```diff
@@ -95,19 +96,70 @@ if(BUILD_TESTING)
     )
     ament_target_dependencies(test_action_client "rcl" "test_msgs")
   endif()
-  ament_add_gtest(test_action_communication
-    test/rcl_action/test_action_communication.cpp
-  )
-  if(TARGET test_action_communication)
-    target_include_directories(test_action_communication PUBLIC
-      include
-      ${rcl_INCLUDE_DIRS}
+
+  # get the rmw implementations ahead of time
+  find_package(rmw_implementation_cmake REQUIRED)
+  get_available_rmw_implementations(rmw_implementations)
+  foreach(rmw_implementation ${rmw_implementations})
+    find_package("${rmw_implementation}" REQUIRED)
+  endforeach()
+
+  function(custom_test_c target)
+    ament_add_gtest(
+      "test_action_communication_${target_suffix}" ${ARGN}
+      TIMEOUT 10
+      APPEND_LIBRARY_DIRS "${append_library_dirs}"
+      ENV
+      RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation}
+      RMW_IMPLEMENTATION=${rmw_implementation}
+      ROS_SECURITY_ROOT_DIRECTORY="${VALID_ROS_SECURITY_ROOT_DIRECTORY}"
+      PATH="${TEST_PATH}"
     )
-    target_link_libraries(test_action_communication
-      ${PROJECT_NAME}
+    if(TARGET test_action_communication_${target_suffix})
+      target_compile_definitions(test_action_communication_${target_suffix}
+        PUBLIC "RMW_IMPLEMENTATION=${rmw_implementation}")
+      target_include_directories(test_action_communication_${target_suffix} PUBLIC
+        include
+        ${rcl_INCLUDE_DIRS}
+      )
+      target_link_libraries(test_action_communication_${target_suffix}
+        ${PROJECT_NAME}
+      )
+      ament_target_dependencies(test_action_communication_${target_suffix}
+        "test_msgs")
+    endif()
+  endfunction()
+
+  macro(targets)
+    set(VALID_ROS_SECURITY_ROOT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/test/test_security_files")
+
+    set(ENV_PATH "$ENV{PATH}")
+    file(TO_CMAKE_PATH "${ENV_PATH}" ENV_PATH)
+    set(TEST_PATH "${ENV_PATH}")
+    if(rmw_implementation STREQUAL "rmw_connext_cpp")
+      if(UNIX AND NOT APPLE)
+        set(RTI_BIN_PATH "$ENV{RTI_OPENSSL_BIN}")
+        file(TO_CMAKE_PATH "${RTI_BIN_PATH}" RTI_BIN_PATH)
+        set(TEST_PATH "${RTI_BIN_PATH};${ENV_PATH}")
+        if(NOT WIN32)
+          string(REPLACE ";" ":" TEST_PATH "${TEST_PATH}")
+        endif()
+      endif()
+    endif()
+
+    if(
+      rmw_implementation STREQUAL "rmw_connext_cpp" OR
```

Comment:
Is there a reason why we only test for these specific implementations? What if we remove this condition?

Reply:
Oops, you're right, not needed at all! That was required for the test_security (from where I took this piece of cmake magic from).
Just updated it, thanks!